pkg/controller/ec2/vpcpeeringconnection/setup.go

```diff
@@ -0,0 +1,141 @@
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"github.com/crossplane/crossplane-runtime/pkg/meta"
+	"github.com/crossplane/crossplane-runtime/pkg/ratelimiter"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+
+	svcapitypes "github.com/crossplane/provider-aws/apis/ec2/v1alpha1"
+	awsclients "github.com/crossplane/provider-aws/pkg/clients"
+
+	"k8s.io/client-go/util/workqueue"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+)
+
+// SetupVPCPeeringConnection adds a controller that reconciles VPCPeeringConnection.
+func SetupVPCPeeringConnection(mgr ctrl.Manager, l logging.Logger, rl workqueue.RateLimiter) error {
+	name := managed.ControllerName(svcapitypes.VPCPeeringConnectionGroupKind)
+	opts := []option{
+		func(e *external) {
+			c := &custom{client: e.client, kube: e.kube}
+			e.postObserve = c.postObserve
+			e.postCreate = c.postCreate
+			e.preCreate = preCreate
+			e.isUpToDate = c.isUpToDate
+			e.filterList = filterList
+		},
+	}
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		WithOptions(controller.Options{
+			RateLimiter: ratelimiter.NewDefaultManagedRateLimiter(rl),
+		}).
+		For(&svcapitypes.VPCPeeringConnection{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(svcapitypes.VPCPeeringConnectionGroupVersionKind),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), opts: opts}),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type custom struct {
+	kube   client.Client
+	client svcsdkapi.EC2API
+}
+
+func filterList(cr *svcapitypes.VPCPeeringConnection, obj *svcsdk.DescribeVpcPeeringConnectionsOutput) *svcsdk.DescribeVpcPeeringConnectionsOutput {
+	connectionIdentifier := aws.String(meta.GetExternalName(cr))
+	resp := &svcsdk.DescribeVpcPeeringConnectionsOutput{}
+	for _, vpcPeeringConnection := range obj.VpcPeeringConnections {
+		if aws.StringValue(vpcPeeringConnection.VpcPeeringConnectionId) == aws.StringValue(connectionIdentifier) {
+			resp.VpcPeeringConnections = append(resp.VpcPeeringConnections, vpcPeeringConnection)
+			break
+		}
+	}
+	return resp
+}
+
+func (e *custom) postObserve(_ context.Context, cr *svcapitypes.VPCPeeringConnection, obj *svcsdk.DescribeVpcPeeringConnectionsOutput, obs managed.ExternalObservation, err error) (managed.ExternalObservation, error) {
+	if err != nil {
+		return managed.ExternalObservation{}, err
+	}
+
+	if awsclients.StringValue(obj.VpcPeeringConnections[0].Status.Code) == "pending-acceptance" && cr.Spec.ForProvider.AcceptRequest {
+		req := svcsdk.AcceptVpcPeeringConnectionInput{
+			VpcPeeringConnectionId: awsclients.String(*obj.VpcPeeringConnections[0].VpcPeeringConnectionId),
+		}
+		request, _ := e.client.AcceptVpcPeeringConnectionRequest(&req)
```

Comment:
NB: Might consider separating the AcceptVpcPeeringConnection into a distinct resource. This would allow accepting connections not generated by crossplane and connections which need different ProviderConfigs (such as those in different accounts). 

A separate resource would avoid having to make a breaking change in the future to support those.

Reply:
i won't get the time to implement another resource :(